lib/Dialect/TTNN/Analysis/L1InterleavedPolicy.cpp

```diff
@@ -24,132 +28,323 @@ uint64_t getOpOutputLayoutUsage(
   return opL1OutputUsage;
 }
 
-void L1InterleavedPolicy::run() {
-  rootOp->walk([&](func::FuncOp func) {
-    DeviceAttr deviceAttr = getCurrentScopeDevice(func);
-    mlir::tt::scheduler::Scheduler scheduler(&func);
-    llvm::SmallVector<mlir::Operation *> scheduleableOps;
-    llvm::DenseMap<Operation *, TTNNLayoutAttr> selectedOpLayout;
-    Operation *currentOp = nullptr;
+L1InterleavedPolicy::Config L1InterleavedPolicy::getOptimalConfig(
+    Operation *baseOp, llvm::DenseMap<Operation *, L1Usage> &opsL1Usage) {
+  uint64_t n, i, j, currentMask;
+  uint64_t currentL1Usage, optimalL1Usage;
+  llvm::DenseMap<Operation *, TTNNLayoutAttr> optimalLayouts;
+  llvm::SmallVector<Operation *> optimalPrecedence;
+
+  optimalL1Usage = 0;
+  n = opsL1Usage.size();
+  for (i = 0; i < (1 << n); i++) {
```

Comment:
Whats this? No comments and it looks like it can overflow.

Reply:
I would also like to see some estimates how will this work in worst case.

Reply:
In what sense do you mean worst case?

Reply:
If an op has `n` operands, the size of `opsL1Usage` will not be larger than `n+1`. Since our ops have at most 2 operands (without including `EmptyOp` as an DPS op's argument), `n` will be at most 3.

Reply:
Add this is comment on assignment to n. Also think about naming it better(than n), like numOperandsBound or something similar.